drivers/bt_ll_nrfxlib/flash/flash_ll_nrfxlib.c

```diff
@@ -366,5 +367,6 @@ static int nrf_btctrl_flash_init(struct device *dev)
 	return 0;
 }
 
-DEVICE_INIT(nrf_btctrl_flash, DT_FLASH_DEV_NAME, nrf_btctrl_flash_init,
-	    NULL, NULL, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+DEVICE_INIT(nrf_btctrl_flash, DT_LABEL(SOC_NV_FLASH_CONTROLLER_NODE),
```

Comment:
why not DT_FLASH_DEV_NAME?

Reply:
DT_FLASH_DEV_NAME was a hack from the dts_fixup.h files. It is gone now.

Previously, every SoC would define it in terms of its own flash controller, like this:

```
#define DT_FLASH_DEV_NAME              DT_LABEL(DT_INST(0, nordic_nrf51_flash_controller))
```

or

```
#define DT_FLASH_DEV_NAME              DT_LABEL(DT_INST(0, nordic_nrf52_flash_controller))
```

and so on.

Now we can just use `flash0` and `flash_controller` node labels in the SoC .dtsi files. This makes each SoC responsible for declaring what node should be used for internal flash, and the C file can be kept short.

This was not possible before the new devicetree.h API, as it is not possible to refer to nodes by node label using the legacy macros.